src/context/internal.rs

```diff
@@ -116,6 +117,37 @@ pub trait ContextInternal<'a>: Sized {
 
     fn activate(&self) { self.scope_metadata().active.set(true); }
     fn deactivate(&self) { self.scope_metadata().active.set(false); }
+
+    fn try_catch_internal<'b: 'a, F>(&mut self, f: F) -> Result<Handle<'a, JsValue>, Handle<'a, JsValue>>
+        where F: FnOnce(&mut Self) -> JsResult<'b, JsValue>
+    {
+        let p = Box::into_raw(Box::new(f)) as *mut c_void;
```

Comment:
I haven't had a chance to mess around with it, but, I _think_ the closure can be unboxed.

Reply:
Yeah, I wasn't confident about this. I'll ask around to find out if extern functions can be safely cast to and from `void *`.

Reply:
OK, found it: [this passage in the unsafe code guidelines](https://rust-lang.github.io/unsafe-code-guidelines/layout/function-pointers.html#representation) indicates that a function can always be safely cast to a pointer:

> The ABI and layout of `(unsafe)? (extern "ABI")? fn(Args...) -> Ret` is exactly that of the corresponding C type -- the lack of a null value does not change this. On common platforms, this means that `*const ()` and `fn(Args...) -> Ret` have the same ABI and layout. This is, in fact, guaranteed by POSIX and Windows.

I double-checked with a Rust core team member, too, so we're good :)

Reply:
Oh wait, this is an `FnOnce` not an `fn`. That might be different… checking with Niko…

Reply:
OK, yeah, a closure is in fact not guaranteed to have any particular layout. I'll add a comment explaining why the boxing is necessary.